content/epmdlessless.md

```diff
@@ -0,0 +1,124 @@
++++
+author = "Tristan Sloughter"
+type="post"
+categories = ["Erlang"]
+date = 2020-11-30T10:41:00Z
+description = ""
+draft = false
+slug = "epmdlessless"
+tags = ["Erlang"]
+title = "Running Erlang Releases without EPMD on OTP 23.1+"
++++
+
+With Erlang/OTP 23.1 and relx 4.2 (included in rebar3 3.14.3) it is now possible
+to conveniently run Erlang releases without relying on EPMD or any third party
+dependency, like [epmdless](https://github.com/tsloughter/epmdless), for opening a
+remote shell or clustering nodes.
+
+The Erlang Port Mapper Daemon (EPMD) is a separate process that is started, if
+not already running, when an Erlang node boots with distribution enabled --
+meaning `-sname` or `-name` is passed to the boot command. The new node [talks to
+the EPMD process](http://erlang.org/doc/apps/erts/erl_dist_protocol.html#epmd-protocol) to register its name with a port that the node will also bind
+to. When a node wants to connect to another it must lookup the port the other
+node is listening on and does so through EPMD. This is required both for
+clustering nodes together and connecting a remote shell on the command line --
+which is essentially the same as clustering because a new node is started before
+attaching to the remote node to bring up a shell on that node.
+
+This has been made possible through a couple new options added to `erl`:
+
+* `-erl_epmd_port P`: Configures the port this node will listen on and use when
+  connecting to remote nodes.
+* `-dist_listen false`: Setting this keeps the node from binding to the port,
+  instead it is only used for connecting to remote nodes. The use case for this
+  is when opening a remote shell `-remsh <node>` (`remote_console` in the
+  release script) to a local node.
+
+And one for [`erl_call`](http://erlang.org/doc/man/erl_call.html) :
```

Comment:
```suggestion
An extra option has also been added to [`erl_call`](http://erlang.org/doc/man/erl_call.html) :
```
People will have lost context after reading the two options, so "one" -> "option" being explicit may be helpful